lib/hexapdf/tokenizer.rb

```diff
@@ -439,9 +440,10 @@ def prepare_string_scanner(needed_bytes = nil)
       @io.seek(@next_read_pos)
       return false if @io.eof?
 
-      @ss << @io.read(8192)
-      if @ss.pos > 8192 && @ss.string.length > 16384
-        @ss.string.slice!(0, 8192)
+      @io.read(8192, @io_partial)
+      @ss << @io_partial
```

Comment:
Nice! Some time ago I have played around with using an output buffer string but didn't find improvements.

I would combine the two lines into one since `IO#read` returns the output string `@io_partial`.

Reply:
Done!